### PR TITLE
VirtualMesh enforce-access-control mesh-specific defaults

### PR DESCRIPTION
**Pull request by harveyxia**
BOT NOTES: 
resolves https://github.com/solo-io/service-mesh-hub/issues/721

### CODE REVIEW

File: cli/pkg/common/table_printing/virtual_mesh_printer.go [hunk 1]
```diff
@@ -25,6 +27,7 @@ type virtualMeshPrinter struct {
 func (m *virtualMeshPrinter) Print(
 	out io.Writer,
 	virtualMeshes []*zephyr_networking.VirtualMesh,
+	meshes []*zephyr_discovery.Mesh,
```
Comment (GrahamGoudeau) [inline cli/pkg/common/table_printing/virtual_mesh_printer.go:30]: so, a couple things here:
1. this is implicitly making the assumption that virtual meshes are homogenous, ie that virtual meshes can be assigned a single mesh type. I'm not sure we should encode that assumption so concretely here
2. I'm not the biggest fan of passing around a list where nil entries are significant/planned for

I'm not actually sure that the printer here should be making the determination of what the concrete enablement status is. If the virtual mesh has indicated "just use the default, whatever that is", then I think this printer should honor whatever the data is on the VirtualMesh, and just output the data as it's presented on the value object.

File: services/mesh-networking/pkg/access/access-control-enforcer/enforcer_defaults.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+const (
+	IstioDefaultAccessControlValue   = false
+	AppmeshDefaultAccessControlValue = true
+)
+
+func DefaultAccessControlValueForMesh(mesh *zephyr_discovery.Mesh) (bool, error) {
+	if mesh == nil {
+		return false, nil
+	}
```
Comment (GrahamGoudeau) [inline services/mesh-networking/pkg/access/access-control-enforcer/enforcer_defaults.go:17]: see other comments on this review, but I'd love to not silently allow nil here, and instead allow it to fall through to the default case below

File: api/networking/v1alpha1/virtual_mesh.proto [hunk 3]
```diff
@@ -89,9 +90,11 @@ message VirtualMeshSpec {
     /*
         If true, by default disallow traffic to all Services in the VirtualMesh unless explicitly allowed through AccessControlPolicies.
         If false, by default allow traffic to all Services in the VirtualMesh.
-        Defaults to false when not set.
+        If omitted, the default value depends on the type of the first mesh listed in `meshes`:
+            Istio: false
+            Appmesh: true
     */
-    bool enforce_access_control = 7;
+    google.protobuf.BoolValue enforce_access_control = 7;
```
[the remark this answers is not in the document]
Comment (ilackarms) [reply api/networking/v1alpha1/virtual_mesh.proto:97]: thoughts on `EnforcementStatus` > `EnforcementPolicy` ?

i prefer to  to reserve "Status" for types that live in the `status` of the cr

[the remark this answers is not in the document]
Comment (harveyxia) [reply api/networking/v1alpha1/virtual_mesh.proto:97]: +1 to both comments

**Pull request merged**